### DISCUSSION TITLE
Keybinds not working

### DISCUSSION THREAD

**Discussion by TheFox666** [VonHeikemen/lsp-zero.nvim#66] [Q&A]
Hi, first of all thanks for this awesome plugin. But i have a big problem with it. It just doesn't set the mappings for me. I tried just using default recommended settings, no change. Now i tried it like in the code below. Also no change. Why isnt my neovim accepting my settings? i Createt a lspzero.lua in my after/plugin folder structure. Can you hint me what i am missing here potentially?

```
local lsp = require('lsp-zero')

lsp.preset('recommended')

lsp.set_preferences({
  set_lsp_keymaps = false
})

lsp.on_attach(function(client, bufnr)
    if vim.b.lsp_attached then return end
    vim.b.lsp_attached = true

    require('nvim-navic').attach(client, bufnr)

    local noremap = { buffer = bufnr, remap = false }
    local bind = vim.keymap.set

    bind('n', '<leader>rn', '<cmd>lua vim.lsp.buf.rename()<cr>', noremap)
end)

lsp.setup()
```

EDIT:
cmp keybinds seem to work, but LSP keybinds just dont exist :(

**Comment by VonHeikemen** [comment]
does it happen with the default keybindings too? Try using `set_lsp_keymaps = true` in `.set_preferences`.

**Reply by TheFox666** [reply]
I also tried just using the minimal setup like this i have no mappings for all the LSP stuff :( 

This is my config if it helps: [Nvim config](https://github.com/TheFox666/.dotfiles/tree/main/nvim/.config/nvim)
Packer Config is at lua/workspace/packer.lua and lspzero is at after/plugin lspzero.lua
```
local lsp = require('lsp-zero')
lsp.preset('recommended')
lsp.setup()
```

**Reply by VonHeikemen** [reply]
Try executing `:LspInfo` and see if the language server is attached to the buffer.

**Reply by TheFox666** [reply]
It is indeed attached i am using neovim 0.9
![image](https://user-images.githubusercontent.com/38280689/205632622-2fadfd2f-35ba-4ed8-a0aa-a1df6a6cc0e0.png)

**Reply by TheFox666** [reply]
I also can see the new hint symbols coming from the plugin also cmp keybinds are as the plugin states, just no lsp mappings

**Reply by VonHeikemen** [reply]
Try executing `:nmap <buffer>` in the buffer with the lsp. it should give you every keymap  in normal mode specific to that buffer.

**Reply by TheFox666** [reply]
No keybinds to see there

**Reply by VonHeikemen** [reply]
How about the command `:LspZeroFormat`? Does that command exists in the buffer?

**Reply by TheFox666** [reply]
Hrm it says "Not an editor command"

**Reply by TheFox666** [reply]
also :verbose nmap gd -> no mapping found

**Reply by TheFox666** [reply]
Looks like the on_attach function of this plugin here isn't going to be called somehow? Also don't have the Command LspZeroWorkspaceRemove

**Reply by VonHeikemen** [reply]
I have a [minimal config](https://github.com/VonHeikemen/lsp-zero.nvim/wiki/Minimal-test-config) in the wiki. Can you test with that?

You can pass the `-u` flag to the `nvim` command to point to a different config.

```sh
nvim -u ./test-config.lua
```

**Reply by TheFox666** [reply]
The minimal test config is working. I had normal lspconfig setup before and removed that for this plugin. Can it be that there is still something cached or something bad with my structure?

**Reply by VonHeikemen** [reply]
maybe you're still calling `lspconfig` somewhere else.

**Reply by TheFox666** [reply]
Nothing, at all :(

**Reply by VonHeikemen** [reply]
do you have your config with lsp-zero on github?

**Reply by TheFox666** [reply]
I am remaking it from scratch there is a plugin lsp-signature which disturbed this one it seems. When i added it nothing worked anymore have to look into it why

**Reply by VonHeikemen** [reply]
Is it this [lsp_signature.nvim](https://github.com/ray-x/lsp_signature.nvim)?

Use the `.on_attach` function they provide.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

local lsp_signature_config = {
  --- configure lsp_signature here
}

lsp.on_attach(function(client, bufnr)
  require('lsp_signature').on_attach(lsp_signature_config, bufnr)
end)

lsp.setup()
```

They are overriding [a builtin neovim function](https://github.com/ray-x/lsp_signature.nvim/blob/9a7e5a093a58b2eefcf77f4a84b8d8c274725d0f/lua/lsp_signature/init.lua#L996) in their setup function. I don't know if that causes the problem, just looks weird to me.

**Reply by TheFox666** [reply]
Yeah thats the plugin, i have to fiddle around with it. thanks for the hint with the code. I try it out. 
Can i also override the keybinds of luasnip with this method?

**Reply by VonHeikemen** [reply]
Do you mean the keybindings to navigate the snippet placeholder? You can override that with `.setup_nvim_cmp`. I have an example in [advance usage page](https://github.com/VonHeikemen/lsp-zero.nvim/blob/main/advance-usage.md#changing-the-keybindings).
